File: frontend/src/api/editor.ts
/**
 * 上級者エディタ（Studio Mode）API クライアント
 */
import {
  AssistRequest,
  AssistResponse,
  AskBibleRequest,
  AskBibleResponse,
  ConsistencyAuditRequest,
  ConsistencyAuditResponse,
  NextBeatsRequest,
  NextBeatsResponse,
  AuditFastHybridRequest,
  UnifiedAuditReport,
} from "../types/editor";
import { apiFetch, handleResponse as handleApiResponse } from "./client";

const BASE = "/api/editor";

/**
 * インライン AI アシスト（五感描写拡張・Show Don't Tell・トーン書き換え）
 */
export async function assistContent(input: AssistRequest): Promise<AssistResponse> {
  const res = await apiFetch(`${BASE}/assist`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(input),
  });
  return handleApiResponse<AssistResponse>(res);
}

/**
 * GraphRAG 専属 AI 編集者への世界観・過去章 Q&A
 */
export async function askBible(input: AskBibleRequest): Promise<AskBibleResponse> {
  const res = await apiFetch(`${BASE}/ask-bible`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(input),
  });
  return handleApiResponse<AskBibleResponse>(res);
}

/**
 * 執筆テキストと設定情報のリアルタイム矛盾診断
 */
export async function auditConsistency(input: ConsistencyAuditRequest): Promise<ConsistencyAuditResponse> {
  const res = await apiFetch(`${BASE}/audit-consistency`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(input),
  });
  return handleApiResponse<ConsistencyAuditResponse>(res);
}

/**
 * Next Beats 3バリエーション（王道・サスペンス・心情）並列生成
 */
export async function generateNextBeats(input: NextBeatsRequest): Promise<NextBeatsResponse> {
  const res = await apiFetch(`${BASE}/next-beats`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(input),
  });
  return handleApiResponse<NextBeatsResponse>(res);
}

/**
 * 矛盾検出された課題の解決・伏線化・例外登録
 */
export async function resolveIssue(
  issueId: string | number,
  action: "Auto-Fix" | "Foreshadowing" | "Ignore",
  apiKey: string = "default-key"
): Promise<{ status: string; message: string }> {
  const res = await apiFetch(`/api/issues/${issueId}/resolve`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      "x-api-key": apiKey,
    },
    body: JSON.stringify({ action }),
  });
  return handleApiResponse<{ status: string; message: string }>(res);
}

/**
 * v5.0: 二層ハイブリッド監査（静的ルール解析＋定性判定）
 */
export async function runHybridAudit(input: AuditFastHybridRequest): Promise<UnifiedAuditReport> {
  const res = await apiFetch(`${BASE}/audit`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(input),
  });
  return handleApiResponse<UnifiedAuditReport>(res, "二層ハイブリッド監査の実行に失敗しました");
}

File: src/backend/routers/editor.py
"""上級者エディタ用 API ルーターモジュール.

インラインAI（五感拡張・Show Don't Tell・リライト）、
GraphRAG 専属AI編集者（Ask Bible・矛盾診断）、
Next Beats 3バリエーション分岐生成エンドポイントを提供。
"""

from __future__ import annotations

import logging

from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from src.backend.database import get_db
from src.models.editor import (
    AskBibleRequest,
    AskBibleResponse,
    AssistRequest,
    AssistResponse,
    ConsistencyAuditRequest,
    ConsistencyAuditResponse,
    NextBeatsRequest,
    NextBeatsResponse,
)
from src.services.editor_assist_service import EditorAssistService
from src.services.editorial_assistant_service import EditorialAssistantService
from src.services.next_beats_service import NextBeatsService

logger = logging.getLogger(__name__)

router = APIRouter(prefix="/api/editor", tags=["editor"])

# サービスインスタンスの初期化
assist_service = EditorAssistService()
editorial_service = EditorialAssistantService()
next_beats_service = NextBeatsService()


@router.post("/assist", response_model=AssistResponse)
async def assist_content(req: AssistRequest) -> AssistResponse:
    """選択テキストに対するインラインAI推敲・五感描写拡張・Show Don't Tell・トーン変換"""
    try:
        return await assist_service.assist(req)
    except Exception as e:
        logger.error(f"Error in assist_content endpoint: {e}")
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/ask-bible", response_model=AskBibleResponse)
async def ask_bible(
    req: AskBibleRequest,
    session: Session = Depends(get_db),
) -> AskBibleResponse:
    """GraphRAG（ベクトル検索 + ナレッジグラフ）を活用した世界観設定資料・過去章 Q&A"""
    try:
        return await editorial_service.ask_bible(session, req)
    except Exception as e:
        logger.error(f"Error in ask_bible endpoint: {e}")
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/audit-consistency", response_model=ConsistencyAuditResponse)
async def audit_consistency(
    req: ConsistencyAuditRequest,
    session: Session = Depends(get_db),
) -> ConsistencyAuditResponse:
    """執筆中の本文と GraphRAG 設定情報とのリアルタイム矛盾診断"""
    try:
        return await editorial_service.audit_consistency(session, req)
    except Exception as e:
        logger.error(f"Error in audit_consistency endpoint: {e}")
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/next-beats", response_model=NextBeatsResponse)
async def generate_next_beats(req: NextBeatsRequest) -> NextBeatsResponse:
    """直前までの本文から、王道・サスペンス・心情の3つの展開バリエーションを並列生成"""
    try:
        return await next_beats_service.generate_three_beats(req)
    except Exception as e:
        logger.error(f"Error in generate_next_beats endpoint: {e}")
        raise HTTPException(status_code=500, detail=str(e))
